client/client-lib/src/lib.rs

```diff
@@ -1102,6 +1102,56 @@ impl Client<UserClientConfig> {
         Ok(confirmed)
     }
 
+    /// Lists all confirmed invoices that have not expired
+    pub async fn list_unexpired_confirmed_invoices(&self) -> Vec<(ContractId, ConfirmedInvoice)> {
+        self.context
+            .db
+            .begin_transaction()
+            .await
+            .find_by_prefix(&ConfirmedInvoiceKeyPrefix)
+            .await
+            .filter_map(
+                |(ConfirmedInvoiceKey(contract_id), confirmed_invoice)| async move {
+                    if !confirmed_invoice.invoice.is_expired() {
```

Comment:
Just because an invoice expired doesn't mean it wasn't paid before that deadline and we checked it since then.

Reply:
Ah yes, I see. I think the idea of checking for expired invoices was as a simpler alternative to saving in the DB whether the invoice was paid, but perhaps we should go with the DB approach instead (see also https://github.com/fedimint/fedimint/issues/1463#issuecomment-1444509118).

Reply:
No problem at all. I've removed the filtering now.

Note that in case `fetch` is run when there are no new contracts to claim, a "transaction already processed" error will be returned for each previously claimed contract, which could grow large over time if there is no filtering on the contracts we attempt to claim. Perhaps I need to change the error handling, but I guess the command will typically be run when there is a new contract to claim.